git-cola: 4.5.0 -> 4.6.1

bobvanderlinden opened the pull request:
## Description of changes

Upgrade of git-cola.

`pyproject = true;` is needed because `setup.py` was removed from the project. See https://github.com/git-cola/git-cola/commit/24c343c88b556af3dc125385218c068894d0b9fd

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bobvanderlinden committed:
git-cola: 4.5.0 -> 4.6.1

bobvanderlinden commented:
Result of `nixpkgs-review pr 294807` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>git-cola</li>
    <li>git-cola.dist</li>
  </ul>
</details>

bobvanderlinden commented:
Result of `nixpkgs-review pr 294807` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>git-cola</li>
    <li>git-cola.dist</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/version-management/git-cola/default.nix:
```diff
@@ -1,21 +1,31 @@
+  nativeBuildInputs = with python3Packages; [ setuptools-scm gettext qt5.wrapQtAppsHook polib ];
   nativeCheckInputs = with python3Packages; [ git pytestCheckHook ];
 
+  pyproject = true;
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/applications/version-management/git-cola/default.nix:
```diff
@@ -1,21 +1,31 @@
 python3Packages.buildPythonApplication rec {
   pname = "git-cola";
-  version = "4.5.0";
+  version = "4.6.1";
```
```suggestion
  version = "4.6.1";  
  pyproject = true;

```
This usually gets placed near the beginning

SuperSandro2000 commented on pkgs/applications/version-management/git-cola/default.nix:
```diff
@@ -1,21 +1,31 @@
   };
 
   buildInputs = lib.optionals stdenv.isLinux [ qt5.qtwayland ];
   propagatedBuildInputs = with python3Packages; [ git pyqt5 qtpy send2trash ];
-  nativeBuildInputs = with python3Packages; [ setuptools-scm gettext qt5.wrapQtAppsHook ];
+  nativeBuildInputs = with python3Packages; [ setuptools-scm gettext qt5.wrapQtAppsHook polib ];
   nativeCheckInputs = with python3Packages; [ git pytestCheckHook ];
```
This would be a lot easier to parse for humans, if we followed the new style used in python packages where between each inputs there is a new line and we write one package per line.

SuperSandro2000 commented on pkgs/applications/version-management/git-cola/default.nix:
```diff
@@ -1,21 +1,31 @@
   buildInputs = lib.optionals stdenv.isLinux [ qt5.qtwayland ];
   propagatedBuildInputs = with python3Packages; [ git pyqt5 qtpy send2trash ];
-  nativeBuildInputs = with python3Packages; [ setuptools-scm gettext qt5.wrapQtAppsHook ];
+  nativeBuildInputs = with python3Packages; [ setuptools-scm gettext qt5.wrapQtAppsHook polib ];
```
We should add setuptools here, to not rely on setuptools-scm propagating it.

bobvanderlinden committed:
git-cola: format file

bobvanderlinden commented:
@SuperSandro2000 thanks for the review! I've reformatted the file so that inputs are on individual lines. I've moved `pyproject` and added setuptools to propagatedBuildInputs. I didn't realize before, but previously I added `polib` as a nativeBuildInput, but since it is a python package probably in use at runtime I moved it to be a propagatedBuildInput as well.

Gave it another test run and could make the commits in this PR with `result/bin/cola` :+1:

bjornfor commented:
Are the following commits fixups to the version bump?
> [git-cola: add polib dependency](https://github.com/NixOS/nixpkgs/pull/294807/commits/7e8b53794de4f74e837833856a0ba687cc8393a2)
> [git-cola: add setuptools propagated build input](https://github.com/NixOS/nixpkgs/pull/294807/commits/82653799cc5ec8cc6baababe8a3e35d9b2b04175)

If they are, please squash, and if they are not, can you say something in the commit message about why they're needed?

bobvanderlinden commented:
> Are the following commits fixups to the version bump?
> 
> > [git-cola: add polib dependency](https://github.com/NixOS/nixpkgs/pull/294807/commits/7e8b53794de4f74e837833856a0ba687cc8393a2)
> > [git-cola: add setuptools propagated build input](https://github.com/NixOS/nixpkgs/pull/294807/commits/82653799cc5ec8cc6baababe8a3e35d9b2b04175)
> 
> If they are, please squash, and if they are not, can you say something in the commit message about why they're needed?

Good point. I've squashed them :+1:

bjornfor reviewed (approved):
Builds and runs.
